framework/include/samplers/Sampler.h

```diff
@@ -18,168 +18,149 @@
-   */
-  std::vector<DenseMatrix<Real>> getSamples();
-
-  /**
-   * Return the sample names, by default 'sample_0, sample_1, etc.' is used.
-   * @return The names assigned to the DenseMatrix items returned by getSamples().
-   *
-   * If custom names are required then utilize the setSampleNames method.
-   */
-  const std::vector<std::string> & getSampleNames() const { return _sample_names; }
-
-  /**
-   * Set the sample names.
-   * @param names List of names to assign to the DenseMatrix entries returned by
-   *              getSamples() method.
+   * Return the number of samples.
+   * @return The total number of rows that exist in all DenseMatrix values from the
+   * getSamples/getLocalSamples methods.
    */
-  void setSampleNames(const std::vector<std::string> & names);
+  dof_id_type getNumberOfRows() const;
+  dof_id_type getNumberOfCols() const;
+  dof_id_type getNumberOfLocalRows() const;
 
+  ///@{
   /**
-   * Store the state of the MooseRandom generator so that new calls to getSamples will create
-   * new numbers.
+   * Return the beginning/end local row index for this processor
    */
-  void execute();
+  dof_id_type getLocalRowBegin() const;
+  dof_id_type getLocalRowEnd() const;
+  ///@}
 
+protected:
+  ///@{
   /**
-   * Return the Sample::Location for the given multi app index.
-   * @param global_index The global row, which is the row if all the DenseMatrix objects were
-   * stacked in order in a single object. When using SamplerTransientMultiApp or
-   * SamplerFullSolveMultiApp the global_index is the MultiApp global index.
-   * @return The location which includes the DenseMatrix index and the row within that matrix.
+   * These methods should be called within the constructor of child classes to define the size of
+   * the matrix to be created.
    */
-  Sampler::Location getLocation(dof_id_type global_index);
+  void setNumberOfRows(dof_id_type n_rows);
+  void setNumberOfCols(dof_id_type n_cols);
+  ///@}
 
   /**
-   * Return the number of samples.
-   * @return The total number of rows that exist in all DenseMatrix values from getSamples()
+   * Set the number of seeds required by the sampler. The Sampler will generate
+   * additional seeds as needed. This function should be called in the constructor
+   * of child objects.
+   * @param number The required number of random seeds, by default this is called with 1.
    */
-  dof_id_type getTotalNumberOfRows();
+  void setNumberOfRandomSeeds(std::size_t number);
 
   /**
-   * Return the number of rows local to this processor.
+   * Get the next random number from the generator.
+   * @param index The index of the seed, by default this is zero. To add additional seeds
+   *              indices call the setNumberOfRequiedRandomSeeds method.
+   *
+   * @return A double for the random number, this is double because MooseRandom class uses double.
    */
-  dof_id_type getLocalNumerOfRows();
+  double rand(unsigned int index = 0);
```

Comment:
could you change the function name to differentiate c++'s own rand()?

Reply:
Yes, what do you suggest?

Reply:
How about `getRand`?

Reply:
great